.pre-commit-config.yaml

```diff
@@ -73,3 +73,42 @@ repos:
         additional_dependencies:
           - sphinx>=1.8.1
           - tomli
+  - repo: https://github.com/adrienverge/yamllint
+    rev: v1.29.0
+    hooks:
+      - id: yamllint
+        args: ["--strict", "--config-file=.yamllint.yml"]
+  - repo: https://github.com/python-jsonschema/check-jsonschema
+    rev: 0.28.0
+    hooks:
+      # TODO: Re-enable this when https://github.com/microsoft/azure-pipelines-vscode/issues/567 is fixed.
+      # - id: check-azure-pipelines
+      - id: check-dependabot
+      - id: check-github-workflows
+      # NOTE: If any of the below schema files need to be changed, be sure to
+      # update the `ci/vendor_schemas.py` script.
+      - id: check-jsonschema
+        files: ^\.appveyor\.yml$
+        args: ["--verbose", "--schemafile", "ci/schemas/appveyor.json"]
+      - id: check-jsonschema
+        files: ^\.circleci/config\.yml$
+        args: ["--verbose", "--schemafile", "ci/schemas/circleciconfig.json"]
+      - id: check-jsonschema
+        files: ^\.github/FUNDING\.yml$
+        args: ["--verbose", "--schemafile", "ci/schemas/github-funding.json"]
+      - id: check-jsonschema
+        files: ^\.github/ISSUE_TEMPLATE/config\.yml$
+        args: ["--verbose", "--schemafile", "ci/schemas/github-issue-config.json"]
+      - id: check-jsonschema
+        files: ^\.github/ISSUE_TEMPLATE/.*\.yml$
+        exclude: ^\.github/ISSUE_TEMPLATE/config\.yml$
+        args: ["--verbose", "--schemafile", "ci/schemas/github-issue-forms.json"]
+      - id: check-jsonschema
+        files: ^\.github/codecov\.yml$
+        args: ["--verbose", "--schemafile", "ci/schemas/codecov.json"]
+      - id: check-jsonschema
+        files: ^\.github/labeler\.yml$
+        args: ["--verbose", "--schemafile", "ci/schemas/pull-request-labeler-5.json"]
+      - id: check-jsonschema
+        files: ^environment\.yml$
+        args: ["--verbose", "--schemafile", "ci/schemas/conda-environment.json"]
```

Comment:
I would prefer if each of these could be given a unique [`name`](https://pre-commit.com/#config-name) that would make it easier to tell what to look for when it fails (I don't actually know what all is printed on failure, but the wall of 8 generic `Validate files with jsonschema` doesn't actually tell me what it has validated (granted, mostly will be skipped when running locally since they all have file filters, but still)

(Also that it says `jsonschema` is mildly confusing since these are yaml files... I understand that the schema is specified in json, but not sure I would remember(/expect if I didn't know this PR) that when the check fails)